src/zimscraperlib/zim/creator.py

```diff
@@ -107,54 +105,72 @@ def __init__(
         self.workaround_nocancel = workaround_nocancel
         self.ignore_duplicates = ignore_duplicates
 
+    def set_mainpath(self, main_path):
+        super().set_mainpath(main_path)
+
     def start(self):
         super().__enter__()
 
-        if getattr(self, "main_path", None):
-            self.set_mainpath(self.main_path)
-
-        if not self.__all_metadata_is_set:
+        if not self._all_metadata_is_set:
             raise ValueError("Mandatory metadata are not all set.")
-        self.__add_metadata(**self.metadata)
+        self.add_metadata(**self._metadata)
+        self.add_default_illustration(self._metadata["Illustration_48x48@1"])
 
         return self
 
-    def __add_metadata(self, **kwargs):
+    def add_metadata(self, **kwargs) -> None:
+        """
+        Use this function to add additional Metadata, please note that the added Metadata should not be duplicated.
+        """
         if kwargs:
-            for name, value in kwargs.items():
-                if value:
-                    self.add_metadata(name, value)
+            for name, value in self._metadata.items():
+                # Illustration needs to be added at the end of start()
+                if name != "Illustration_48x48@1" and value:
+                    self._validate_metadata(name, value)
+                    super().add_metadata(name, value)
+
+    def _validate_metadata(self, name, value):
```

Comment:
We should also call this in an override of `add_metadata`

Reply:
We already call this in the `add_metadata` here, see line 129.

Reply:
As explained, we don't want this version of add_metadata